tests/e2e/specs/wcpay/shopper/shopper-checkout-purchase-site-editor.spec.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+
+const {
+	shopper,
+	merchant,
+	activateTheme,
+} = require( '@woocommerce/e2e-utils' );
+
+/**
+ * Internal dependencies
+ */
+
+import {
+	fillCardDetails,
+	confirmCardAuthentication,
+	setupProductCheckoutNoMiniCart,
+} from '../../../utils/payments';
+
+import { shopperWCP, merchantWCP } from '../../../utils';
+
+const cardTestingPreventionStates = [
+	{ cardTestingPreventionEnabled: false },
+	{ cardTestingPreventionEnabled: true },
+];
+
+describe.each( cardTestingPreventionStates )(
+	'Successful purchase, site builder theme',
+	( { cardTestingPreventionEnabled } ) => {
+		beforeAll( async () => {
+			await merchant.login();
+			await activateTheme( 'twentytwentyfour' );
+			if ( cardTestingPreventionEnabled ) {
+				await merchantWCP.enableCardTestingProtection();
+			}
+			await merchant.logout();
+		} );
+
+		beforeEach( async () => {
+			await setupProductCheckoutNoMiniCart(
```

Comment:
Can we add `shopperWCP.emptyCart()` on `beforeEach` to ensure the test environment is properly setup

Reply:
Added in https://github.com/Automattic/woocommerce-payments/pull/8164/commits/a07d017aa9c3852e1b23538fece0319c4c24b285.